include/envoy/filesystem/filesystem.h

```diff
@@ -14,58 +14,62 @@ namespace Envoy {
 namespace Filesystem {
 
 /**
- * Abstraction for a file on disk.
+ * Abstraction for a basic file on disk.
  */
-class File {
+class RawFile {
```

Comment:
From an interface perspective, I'm not really sure what the difference is between a "raw" file and a "file." Can you at minimum add more comments and maybe think of a more descriptive name for `File` below?

Reply:
+1 for more comments. From my perspective a RawFile is an OS abstraction, and a File (maybe a better name would be good) adds Envoy semantics like stats and auto-flush.

RawFile needs to be re-implemented for different physical file layers, but the stats & auto-flush functionality can be shared across multiple RawFile impls.

Reply:
As far as I can tell, the only use of the `File` class is by the AccessLogManager, so maybe `AccessLogFile` or something like that might be better?

Reply:
If we leave it here I would probably do something like `ThreadSafeFile` or `ThreadSafeAutoFlushFile` or something. One other option is to actually completely move this file type and its creation into the access log manager interface, and then the filesystem interface just knows how to make "raw" or basic files, and then we can just call it `File`. WDYT?

Reply:
I think moving it into the access log manager interface makes sense -- we'll give it a look